src/xform_to_datalog/manifest_datalog_facts.cc

```diff
@@ -33,24 +33,14 @@ namespace types = raksha::ir::types;
 //  traversal. This works, but it is hard to read and hard to test. We should
 //  break this up.
 ManifestDatalogFacts ManifestDatalogFacts::CreateFromManifestProto(
-    const arcs::ManifestProto &manifest_proto,
-    ir::ParticleSpecRegistry &particle_spec_registry) {
+    const ir::SystemSpec* system_spec,
+    const arcs::ManifestProto &manifest_proto) {
```

Comment:
See other comments elsewhere, but I think we should use a `const ir::SystemSpec &` to ensure `system_spec` is non-null.